src/pixelator/pixeldataset/precomputed_layouts.py

```diff
@@ -253,81 +387,94 @@ def data():
-    add_node_marker_counts,
-    layout_algorithm,
-    all_markers,
+    add_node_marker_counts: bool,
+    layout_algorithms: list[SupportedLayoutAlgorithm] | SupportedLayoutAlgorithm,
+    all_markers: list[str],
 ):
+    if isinstance(layout_algorithms, str):
+        layout_algorithms = [layout_algorithms]
+
     def data():
         for component, df in edgelist.group_by("component"):
             logger.debug("Computing for component %s", component)
-            yield (
-                pl.DataFrame(
-                    Graph.from_edgelist(
-                        df.lazy(),
-                        add_marker_counts=add_node_marker_counts,
-                        simplify=True,
-                        use_full_bipartite=True,
+            graph = Graph.from_edgelist(
+                df.lazy(),
+                add_marker_counts=add_node_marker_counts,
+                simplify=True,
+                use_full_bipartite=True,
+            )
+            for layout_algorithm in layout_algorithms:
+                yield (
+                    pl.DataFrame(
+                        graph.layout_coordinates(
+                            layout_algorithm=layout_algorithm,
+                            get_node_marker_matrix=add_node_marker_counts,
+                            cache=False,
+                        )
+                        .pipe(
+                            _zero_fill_missing_markers
+                            # If we don't want to add the marker counts
+                            # we just pass this through an identity function
+                            if add_node_marker_counts
+                            else lambda df, all_markers: df,
+                            all_markers=all_markers,
+                        )
+                        .sort_index(axis=1)
                     )
-                    .layout_coordinates(
-                        layout_algorithm=layout_algorithm,
-                        get_node_marker_matrix=add_node_marker_counts,
-                        cache=False,
+                    .with_columns(
+                        component=pl.lit(component),
+                        graph_projection=pl.lit("bipartite"),
+                        layout=pl.lit(layout_algorithm),
                     )
-                    .pipe(_zero_fill_missing_markers, all_markers=all_markers)
-                    .sort_index(axis=1)
-                ).with_columns(
-                    component=pl.lit(component),
-                    graph_projection=pl.lit("bipartite"),
-                    layout=pl.lit(layout_algorithm),
+                    .lazy()
                 )
-            )
 
-    result = pl.concat(data())
+    result = pl.concat(data(), how="diagonal")
     return result
 
 
-def _fn(d):
-    (edgelist, add_node_marker_counts, layout_algorithm, all_markers) = d
-    return _get_layout(edgelist, add_node_marker_counts, layout_algorithm, all_markers)
+def _wrap_get_layouts(d):
```

Comment:
Wouldn't work *args here?

Reply:
It would, but I wanted to make it clear which inputs `_compute_layouts` expects.